service/matching/taskQueueManager.go

```diff
@@ -630,3 +700,71 @@ func newIOContext() (context.Context, context.CancelFunc) {
 
 	return ctx, cancel
 }
+
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartitionOnInit(ctx context.Context) {
+	if c.metadataInitialFetchFut.Ready() {
+		return
+	}
+	_, err := c.fetchMetadataFromRootPartition(ctx)
+	c.metadataInitialFetchFut.Set(struct{}{}, err)
+}
+
+// fetchMetadataFromRootPartition fetches metadata from root partition iff this partition is not a root partition.
+// Returns the fetched data, if fetching was necessary and successful.
+func (c *taskQueueManagerImpl) fetchMetadataFromRootPartition(ctx context.Context) (*persistencespb.VersioningData, error) {
+	// Nothing to do if we are the root partition
+	// (for versioning - any later added metadata may need to not abort so early)
+	if c.taskQueueID.IsRoot() {
+		return nil, nil
+	}
+
+	rootTqName := c.taskQueueID.GetRoot()
+	curHash := HashVersioningData(c.db.versioningData)
+	if len(curHash) == 0 {
+		// if we have no data, make sure we send a sigil value, so it's known we desire versioning data
+		curHash = []byte{0}
+	}
+	res, err := c.matchingClient.GetTaskQueueMetadata(ctx, &matchingservice.GetTaskQueueMetadataRequest{
+		NamespaceId:               c.taskQueueID.namespaceID.String(),
+		TaskQueue:                 rootTqName,
+		WantVersioningDataCurhash: curHash,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
if the initial request happens to fail, we need to keep trying until we either get some data or get a successful "no data". in other words I think we need the polling loop to just start out running, and maybe as an optimization it can exit early if there's no data. (but then needs to start again on an invalidate)

Reply:
Hmm... do we need to? Right now it failing will mean that `WaitUntilInitialized` will error out, and thus the manager isn't created and so the top level operation wanting it will bail, and we'll get to just try again next time someone wants something